src/commands/HBonds.cpp

```diff
@@ -0,0 +1,206 @@
+                                and <stride> optional. Default is to use all
+                                steps from the input; starting at 0, ending at
+                                the last step, and with a stride of 1.
+  --selection_acceptor=<sel>    selection to use for the acceptors. This must be a
+                                selection of size 2 and type bonds.
+                                [default: bonds: type(#2) == H]
+  --selection_donor=<sel>       selection to use for the donors. This must be a
+                                selection of size 1.
+                                [default: atoms: not type H]
+  -p <par>, --parameters=<par>  parameters to use for the hydrogen bond. <par>
+                                format is <d:α> where 'd' is the donor-acceptor 
+                                maximum distance in angstroms and 'α' is the
+                                donor-acceptor-hydrogen maximum angle in degrees.
+                                [default: 3.0:30.0]
+)";
+
+static HBonds::Options parse_options(int argc, const char* argv[]) {
+    auto options_str = command_header("hbonds", HBonds().description()) + "\n";
+    options_str += OPTIONS;
+    auto args = docopt::docopt(options_str, {argv, argv + argc}, true, "");
+
+    HBonds::Options options;
+    options.trajectory = args.at("<trajectory>").asString();
+    options.guess_bonds = args.at("--guess-bonds").asBool();
+
+    options.selectionAcceptor = args.at("--selection_acceptor").asString();
+    options.selectionDonor = args.at("--selection_donor").asString();
+
+    if (args.at("--output")){
+        options.outfile = args.at("--output").asString();
+    } else {
+        options.outfile = options.trajectory + "_hb.dat";
+    }
+
+    if (args.at("--steps")) {
+        options.steps = steps_range::parse(args.at("--steps").asString());
+    }
+
+    if (args.at("--format")){
+        options.format = args.at("--format").asString();
+    }
+
+    if (args.at("--topology")){
+        if (options.guess_bonds) {
+            throw CFilesError("Can not use both '--topology' and '--guess-bonds'");
+        }
+        options.topology = args.at("--topology").asString();
+    }
+
+    if (args.at("--topology-format")){
+        if (options.topology == "") {
+            throw CFilesError("Useless '--topology-format' without '--topology'");
+        }
+        options.topology_format = args.at("--topology-format").asString();
+    }
+
+    if (args.at("--cell")) {
+        options.custom_cell = true;
+	options.cell = parse_cell(args.at("--cell").asString());
+	}
+
+    options.distance_parameter = 3.0;
+    options.angle_parameter = 30.0;
+    if (args.at("--parameters")) {
+	auto splitted = split(args.at("--parameters").asString(), ':');
+	if (splitted.size() == 2) {
+	    options.distance_parameter = stod(splitted[0]);
+	    options.angle_parameter = stod(splitted[1])*pi/180;
```

Comment:
What happens if `splitted[0] == "hviousbo"` ? (look for `std::invalid_argument` [here](http://en.cppreference.com/w/cpp/string/basic_string/stof))